sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -712,6 +717,109 @@ case class UnresolvedStarExceptOrReplace(
   }
 }
 
+/**
+ * Represents some of the input attributes to a given relational operator, for example in
+ * `df.withColumn`.
+ *
+ * @param target an optional name that should be the target of the expansion. If omitted all
+ *              targets' columns are produced. This can only be a table name. This
+ *              is a list of identifiers that is the path of the expansion.
+ *
+ * @param colNames a list of column names that should be replaced or produced.
+ *
+ * @param exprs a list of expressions that should be used to replace the expressions removed by.
+ *
+ * @param explicitMetadata an optional list of explicit metadata to associate with the columns.
+ */
+case class UnresolvedStarWithColumns(
+     target: Option[Seq[String]],
```

Comment:
This is not used at all in this expression. We should remove it from the parameter list, and just override it to always return `None`